src/emu/emuopts.cpp

```diff
@@ -541,20 +544,66 @@ void emu_options::set_system_name(const char *name)
 {
 	// remember the original system name
 	std::string old_system_name(system_name());
+	bool new_system = old_system_name.compare(name)!=0;
 
 	// if the system name changed, fix up the device options
-	if (old_system_name.compare(name)!=0)
+	if (new_system)
 	{
 		// first set the new name
 		std::string error;
 		set_value(OPTION_SYSTEMNAME, name, OPTION_PRIORITY_CMDLINE, error);
 		assert(error.empty());
 
-		// remove any existing device options and then add them afresh
+		// remove any existing device options
 		remove_device_options();
-		while (add_slot_options()) { }
+	}
+
+	// get the new system
+	const game_driver *cursystem = system();
+	if (cursystem == nullptr)
+		return;
+
+	if (software_name())
```

Comment:
this have been fixed in e3988ad65e5b25dcc956ea43e008f9ea2d3ce329

On Mon, Mar 28, 2016 at 2:35 AM, balr0g <notifications@github.com> wrote:

> In src/emu/emuopts.cpp
> <https://github.com/mamedev/mame/pull/743#discussion_r57537522>:
>
> >  		remove_device_options();
> > -		while (add_slot_options()) { }
> > +	}
> > +
> > +	// get the new system
> > +	const game_driver *cursystem = system();
> > +	if (cursystem == nullptr)
> > +		return;
> > +
> > +	if (software_name())
>
> @ajrhacker <https://github.com/ajrhacker> software_name() appears to
> return an empty C-string in my testing using the MAME builtin UI. This
> means the code inside the if-statement will be executed. Probably a bug?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/mamedev/mame/pull/743/files/dee9b2d34a4826c1b9d83e2b5602c2093f7eb4bc#r57537522>
>